virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -431,39 +503,47 @@ def calculate_litter_additions_from_herbivory(self) -> dict[str, DataArray]:
         }
 
     def calculate_soil_additions(self) -> dict[str, DataArray]:
-        """Calculate the how much animal matter should be transferred to the soil."""
+        """Calculate how much animal matter should be transferred to the soil."""
 
         nutrients = ["carbon", "nitrogen", "phosphorus"]
-        # Find the size of all decomposed excrement and carcass pools
+
+        # Find the size of all decomposed excrement and carcass pools, by cell_id
         decomposed_excrement = {
             nutrient: [
-                community.excrement_pool.decomposed_nutrient_per_area(
+                pool.decomposed_nutrient_per_area(
                     nutrient=nutrient, grid_cell_area=self.data.grid.cell_area
                 )
-                for community in self.communities.values()
+                for cell_id, pools in self.excrement_pools.items()
+                for pool in pools
             ]
             for nutrient in nutrients
         }
+
         decomposed_carcasses = {
             nutrient: [
-                community.carcass_pool.decomposed_nutrient_per_area(
+                pool.decomposed_nutrient_per_area(
                     nutrient=nutrient, grid_cell_area=self.data.grid.cell_area
                 )
-                for community in self.communities.values()
+                for cell_id, pools in self.carcass_pools.items()
+                for pool in pools
             ]
             for nutrient in nutrients
         }
 
-        # All excrement and carcasses in their respective decomposed subpools are moved
-        # to the litter model, so stored carbon of each subpool is reset to zero
-        for community in self.communities.values():
-            community.excrement_pool.decomposed_carbon = 0.0
-            community.excrement_pool.decomposed_nitrogen = 0.0
-            community.excrement_pool.decomposed_phosphorus = 0.0
-            community.carcass_pool.decomposed_carbon = 0.0
-            community.carcass_pool.decomposed_nitrogen = 0.0
-            community.carcass_pool.decomposed_phosphorus = 0.0
+        # Reset all decomposed excrement pools to zero
+        for excrement_pools in self.excrement_pools.values():
+            for excrement_pool in excrement_pools:
+                excrement_pool.decomposed_carbon = 0.0
+                excrement_pool.decomposed_nitrogen = 0.0
+                excrement_pool.decomposed_phosphorus = 0.0
+
+        for carcass_pools in self.carcass_pools.values():
+            for carcass_pool in carcass_pools:
+                carcass_pool.decomposed_carbon = 0.0
+                carcass_pool.decomposed_nitrogen = 0.0
+                carcass_pool.decomposed_phosphorus = 0.0
```

Comment:
I think it will make sense to implement a `reset` method for these pools, so you can do the following without having to know the internal detail:

```suggestion
        for excrement_pools in self.excrement_pools.values():
            for excrement_pool in excrement_pools:
                excrement_pool.reset()

        for carcass_pools in self.carcass_pools.values():
            for carcass_pool in carcass_pools:
                carcass_pool.reset()
```

Reply:
@dalonsoa this is code that I added and Taran had to refactor to get to work with the new setup. I agree that this is a better way of doing things, I'll rewrite and push the changes to this branch